pyproject.toml

```diff
@@ -172,6 +160,27 @@ requires = ["setuptools>=77.0.0",
             "extension-helpers>=1.4,<2"]
 build-backend = "setuptools.build_meta"
 
+[dependency-groups]
+# re-define historical optional dependencies (a.k.a. 'extras') as
+# user-invisible, PEP 735 dependency groups
+# These group together all the dependencies needed for developing in Astropy.
+dev = [
+    "astropy[recommended]",  # installs the most common optional dependencies
+    "astropy[test]",
+    "astropy[docs]",
+    "astropy[typing]",
```

Comment:
Not sure this is completely right. Aren't dependency groups meant to be independent of the base package? This installs Astropy.

Reply:
> Aren't dependency groups meant to be independent of the base package? This installs Astropy.

I don't know that it was a design goal, but you're right, groups normally allow to decouple their installation from the package's, but I don't know of a way to do that here *without* also duplicating data from extras...